include/aspect/particle/property/elastic_stress.h

```diff
@@ -0,0 +1,126 @@
+ along with ASPECT; see the file LICENSE.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect_particle_property_elastic_stress_h
+#define _aspect_particle_property_elastic_stress_h
+
+#include <aspect/particle/property/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/material_model/visco_plastic.h>
+#include <aspect/material_model/viscoelastic.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      /**
+       * A class that calculates the viscoelastic stress a particle
+       * has accumulated through time.
+       * The implementation of this property is equivalent to the implementation
+       * for compositional fields that is described in the viscoelastic material
+       * model.
+       *
+       * @ingroup ParticleProperties
+       */
+      template <int dim>
+      class ElasticStress : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+      {
+        public:
+          /**
+          * Constructor.
+          */
+          ElasticStress ();
+
+          void initialize () override;
+
+          /**
+                 * Initialization function. This function is called once at the
+                 * creation of every particle for every property to initialize its
+                 * value.
+                 *
+                 * @param [in] position The current particle position.
+                 * @param [in,out] particle_properties The properties of the particle
+                 * that is initialized within the call of this function. The purpose
+                 * of this function should be to extend this vector by a number of
+                 * properties.
+                 */
+          void
+          initialize_one_particle_property (const Point<dim> &position,
+                                            std::vector<double> &particle_properties) const override;
+
+          /**
+           * Update function. This function is called every time an update is
+           * request by need_update() for every particle for every property.
+           *
+           * @param [in] data_position An unsigned integer that denotes which
+           * component of the particle property vector is associated with the
+           * current property. For properties that own several components it
+           * denotes the first component of this property, all other components
+           * fill consecutive entries in the @p particle_properties vector.
+           *
+           * @param [in] position The current particle position.
+           *
+           * @param [in] solution The values of the solution variables at the
+           * current particle position.
+           *
+           * @param [in] gradients The gradients of the solution variables at
+           * the current particle position.
+           *
+           * @param [in,out] particle_properties The properties of the particle
+           * that is updated within the call of this function.
```

Comment:
This documentation is out of date, please use the updated one, or use `@copydoc aspect::Particle::Property::update_particle_property()`